File: OCH_Tao/tessoku-book/A14.js
const [[N, K], A, B, C, D] = require("fs").readFileSync("/dev/stdin", "utf8").trim().split("\n").map(x => x.split(" ").map(y => parseInt(y, 10)));
const P = new Set();
const Q = new Set();
for (const i of A) {
  for (const j of B) {
    P.add(i + j);
  }
}
for (const i of C) {
  for (const j of D) {
    Q.add(i + j);
  }
}
for (const i of P) {
  if (Q.has(K - i)) {
    console.log("Yes");
    process.exit();
  }
}
console.log("No");

File: OCH_Tao/tessoku-book/A14.py
N, K = map(int, input().split())
A = list(map(int, input().split()))
B = list(map(int, input().split()))
C = list(map(int, input().split()))
D = list(map(int, input().split()))
P = set()
Q = set()
for i in A:
  for j in B:
    P.add(i+j)
for i in C:
  for j in D:
    Q.add(i+j)
for i in P:
  if K-i in Q:
    print("Yes")
    break
else:
  print("No")
